### PR TITLE
refactored pdf generation codes

### PR DESCRIPTION
**Pull request by akenzua**
[TP11801](https://maps.tpondemand.com/restui/board.aspx#page=board/5682182231901749200&appConfig=eyJhY2lkIjoiNjA5RjRCQzM4NTlDRjBBOEQ4QzM2ODUxMEVENDhFNkUifQ==&boardPopup=userstory/11801/silent)

This PR replaced the old GTA codes with the ones sent by the agency. This however broke the pdf rendering part of the application. The temporary solution sought was to provide a static file for rendering the pdf files and a [ticket](https://maps.tpondemand.com/restui/board.aspx#page=board/5682182231901749200&appConfig=eyJhY2lkIjoiNjA5RjRCQzM4NTlDRjBBOEQ4QzM2ODUxMEVENDhFNkUifQ==&boardPopup=userstory/11810) for the dynamic rendering has been written to be addressed as a tech debt.

### CODE REVIEW

File: pages/_document.js [hunk 1]
```diff
@@ -36,25 +37,37 @@ export default class MyDocument extends Document {
     return (
       <Html dir="ltr">
         <Head>
-          {/* Global Site Tag (gtag.js) - Google Analytics */}
+          <Fragment></Fragment>
           <script
             async
             src={`https://www.googletagmanager.com/gtag/js?id=${GA_TRACKING_ID}`}
           />
```
Comment (davidtrussler) [inline pages/_document.js:44]: I believe this `<script>` element is part of what should be removed for this piece of work.

**Comment by davidtrussler** [comment]
I'm finding this PR very confusing: it is supposed to be about changing some tracking code but is mostly about a different issue. I think it would help if these were dealt with separately. I can't see how the tracking code would break the PDF code so would it be better to deploy this and work on the PDF issue separately?

**Comment by akenzua** [comment]
> I'm finding this PR very confusing: it is supposed to be about changing some tracking code but is mostly about a different issue. I think it would help if these were dealt with separately. I can't see how the tracking code would break the PDF code so would it be better to deploy this and work on the PDF issue separately?

Yes it is confusing. The PDF had its problem before this PR but was made worse after the GTA code was changed probably because of the closing script tag. So i needed to sort the PDF also. What I did with the PDF was a kind of temporary solution to make the app work meanwhile. I created a ticket to solve the problem, which look like the one here, only that the text for the pdf will be generated dynamically.

**Comment by davidtrussler** [comment]
> Yes it is confusing. The PDF had its problem before this PR but was made worse after the GTA code was changed probably because of the closing script tag. So i needed to sort the PDF also. What I did with the PDF was a kind of temporary solution to make the app work meanwhile. I created a ticket to solve the problem, which look like the one here, only that the text for the pdf will be generated dynamically.
> 
Ah right, so if the closing tag is fixed is the PDF issue the same as before? If it is do we still need the PDF stuff in this PR?

File: pages/_document.js (hunk 1 above)
Comment (akenzua) [reply pages/_document.js]: This has been removed

File: pages/_document.js [hunk 2]
```diff
@@ -36,25 +37,37 @@ export default class MyDocument extends Document {
...
-            gtag('js', new Date());
-            gtag('config', '${GA_TRACKING_ID}', {
-              page_path: window.location.pathname,
-            });
+              __html: `<script
+              (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+              new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
+              j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+              'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+              })($ = window,document,'script','dataLayer','GTM-WVFLH9');/>
```
[the remark this answers is not in the document]
Comment (akenzua) [reply pages/_document.js]: closed

**Comment by akenzua** [comment]
> > Yes it is confusing. The PDF had its problem before this PR but was made worse after the GTA code was changed probably because of the closing script tag. So i needed to sort the PDF also. What I did with the PDF was a kind of temporary solution to make the app work meanwhile. I created a ticket to solve the problem, which look like the one here, only that the text for the pdf will be generated dynamically.
> 
> Ah right, so if the closing tag is fixed is the PDF issue the same as before? If it is do we still need the PDF stuff in this PR?

Just did. It didn't fix it. The problem has to do with generating the file on the fly

File: pages/_document.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (davidtrussler) [reply pages/_document.js]: This still enclosed one `<script>` tag inside another: 

image

I think we need something more like: 
```
          <script
            dangerouslySetInnerHTML={{
              __html: `(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
              new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
              j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
              'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
              })($ = window,document,'script','dataLayer','GTM-WVFLH9');`
            }}
          />
```

**Comment by davidtrussler** [comment]
> Just did. It didn't fix it. The problem has to do with generating the file on the fly
> 
OK, let's go with this solution for now then. When the `<script>` tag is fixed I will approve it.

File: pages/_document.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (akenzua) [reply pages/_document.js]: It didnt work that way. You should try it. This seem like the only way to make it work.

[the remark this answers is not in the document]
Comment (davidtrussler) [reply pages/_document.js]: > It didnt work that way. You should try it. This seem like the only way to make it work.
> 
I'm getting this, which looks right to me: 
image

**Review by ladamsatmaps19 — APPROVED** [review]
Read all comments. Thanks @davidtrussler and @akenzua. Happy to approve

File: Utils/layouts/head.js [hunk 3]
```diff
@@ -14,6 +13,23 @@ const AppHead = ({ path, t }) => {
     pageTitle = t("head.page.landing");
   }
 
+  const $ = "window";
```
Comment (davidtrussler) [inline Utils/layouts/head.js:16]: I'm not sure this is right: I'm pretty sure the `$` is standing in for JQuery. You can replace `window` with any value and the result is the same as far as I can see.

File: Utils/layouts/head.js [hunk 4]
```diff
@@ -14,6 +13,23 @@ const AppHead = ({ path, t }) => {
...
+
+  const script = process.browser ? (
```
Comment (davidtrussler) [inline Utils/layouts/head.js:18]: Likewise I'm not convinced we need to stop this script being added server-side. I don't believe it runs anything on load. I believe also that the isomorphic nature of the app will cause problems whenever the server-side and hydrated client-side code differ.

Comment (akenzua) [reply Utils/layouts/head.js:16]: No that wasnt the solution, i found. I have removed the` const $ = window`

**Review by davidtrussler — APPROVED** [review]
I believe this is looking good now - 👍

**Review by davidtrussler — APPROVED** [review]
The updates on this fix the issues. Good work! 👍

**Pull request merged**